### PR TITLE
Update mac Jenkins agent script

### PR DESCRIPTION
**Pull request by MialLewis**
#### Summary of work

Since a recent Jenkins update, the older mac Jenkins agents stopped working. This PR updates the script to download new Jenkins agents from a different location.

Also removes depreciated terminology.

### To test:

Code review - check status of NDW1170, as this script is currently running on there.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: buildconfig/Jenkins/jenkins-agent.sh [hunk 1]
```diff
@@ -91,12 +96,12 @@ elif [ -f ${HOME}/Jenkins/${JAR_FILE} ]; then
 else
   JAR_FILE_TMP=/tmp/${JAR_FILE}
   if [ ! -f ${JAR_FILE_TMP} ]; then
-    echo "Downloading slave jar file to ${JAR_FILE_TMP}"
+    echo "Downloading agent jar file to ${JAR_FILE_TMP}"
     if [ $(command -v curl) ]; then
-      echo "curl --location -o ${JAR_FILE_TMP} ${JENKINS_URL}/jnlpJars/${JAR_FILE}"
-      curl -o ${JAR_FILE_TMP} ${JENKINS_URL}/jnlpJars/${JAR_FILE}
+      echo "curl --location -o ${JAR_FILE_TMP} ${JENKINS_REPO_URL}/${JAR_VERSION}/${JAR_FILE}"
```
Comment (MohamedAlmaki) [inline buildconfig/Jenkins/jenkins-agent.sh:101]: Just for my knowledge, could you explain why are you replacing the old link with one directly from the Jenkins repo?

File: buildconfig/class_maker.py [hunk 2]
```diff
@@ -184,7 +184,7 @@ def write_test(subproject, classname, filename, args):
   void test_exec()
   {{
     // Create test input if necessary
-    MatrixWorkspace_sptr inputWS = //-- Fill in appropriate code. Consider using TestHelpers/WorkspaceCreationHelpers.h --
+    MatrixWorkspace_sptr inputWS = //-- Fill in appropriate code. Consider using MantidFrameworkTestHelpers/WorkspaceCreationHelper.h --
```
Comment (MohamedAlmaki) [inline buildconfig/class_maker.py:188]: How is this change related to the current PR?

File: buildconfig/Jenkins/jenkins-agent.sh (hunk 1 above)
Comment (MialLewis) [reply buildconfig/Jenkins/jenkins-agent.sh:101]: In the past, the Jenkins agents provided via our Jenkins server were always compatible with the version of Jenkins we were running. For some reason, this is now not the case. We have to go directly to the Jenkins repo and find an appropriate version there.

This could perhaps be because Jenkins are moving away from distributing agents in this way, and the preferred solution is to use agents baked into docker containers (doesn't help us on mac!).

File: buildconfig/class_maker.py (hunk 2 above)
Comment (MialLewis) [reply buildconfig/class_maker.py:188]: This is an unrelated change I was making previously that snuck in with no relevance, It's still valid though. Can remove it if you want.

File: buildconfig/Jenkins/jenkins-agent.sh (hunk 1 above)
Comment (MohamedAlmaki) [reply buildconfig/Jenkins/jenkins-agent.sh:101]: Ok, thanks for the explanation

File: buildconfig/class_maker.py (hunk 2 above)
Comment (MohamedAlmaki) [reply buildconfig/class_maker.py:188]: Ok, no problem, I will approve it

**Review by MohamedAlmaki — APPROVED** [review]
LGTM
